bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -0,0 +1,265 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.discovery.addon.ip;
+
+import static org.openhab.core.config.discovery.addon.AddonFinderConstants.SERVICE_NAME_IP;
+import static org.openhab.core.config.discovery.addon.AddonFinderConstants.SERVICE_TYPE_IP;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.StandardProtocolFamily;
+import java.net.StandardSocketOptions;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.nio.channels.DatagramChannel;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.text.ParseException;
+import java.util.HashSet;
+import java.util.HexFormat;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.StringTokenizer;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.addon.AddonDiscoveryMethod;
+import org.openhab.core.addon.AddonInfo;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.config.discovery.addon.AddonFinder;
+import org.openhab.core.config.discovery.addon.BaseAddonFinder;
+import org.openhab.core.net.NetUtil;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a {@link IpAddonFinder} for finding suggested add-ons by sending IP packets to the
+ * network and collecting responses.
+ * 
+ * @implNote On activation, a thread is spawned which handles the detection. Scan runs once,
+ *           no continuous background scanning.
+ *
+ * @author Holger Friedrich - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = AddonFinder.class, name = IpAddonFinder.SERVICE_NAME)
+public class IpAddonFinder extends BaseAddonFinder {
+
+    public static final String SERVICE_TYPE = SERVICE_TYPE_IP;
+    public static final String SERVICE_NAME = SERVICE_NAME_IP;
+
+    private final Logger logger = LoggerFactory.getLogger(IpAddonFinder.class);
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(SERVICE_NAME);
+    private @Nullable Future<?> scanJob = null;
+    Set<AddonInfo> suggestions = new HashSet<>();
+
+    public IpAddonFinder() {
+        logger.warn("IpAddonFinder::IpAddonFinder");
+    }
+
+    @Deactivate
```

Comment:
I would expect an @Activate method as well and a regular (scheduled) poll. Remember this should be not just for KNX. The whole finder can be switched off. If it is not switched off, it should poll for all defined ip broadcast finders on a schedule.

Reply:
This should also be addressed in a 4.2 PR. I understand the code to only do a scan on start of the finder or added AddonInfo. I would expect a regular scan. I think this is fine now, but I would like to see it improved in the future.